src/service/resources-impl.js

```diff
@@ -1632,8 +1668,16 @@ export class Resource {
     } catch (e) {
       return Promise.reject(e);
     }
+
     this.layoutPromise_ = promise.then(() => this.layoutComplete_(true),
         reason => this.layoutComplete_(false, reason));
+
+    if (!this.whenFirstLayoutCompletePromise_) {
```

Comment:
This won't work. In this case, the first call to `whenFirstLayoutComplete()` will return NULL. Instead, you need to move this code to the `whenFirstLayoutComplete()` itself. And in this method you simply need:
```
if (this.whenFirstLayoutCompleteResolve_) {
  this.layoutPromise_.then(this.whenFirstLayoutCompleteResolve_);
}
```

This is also a good test case that it'd be worth to add to the tests. I.e. call `whenFirstLayoutComplete()` before first `startLayout` has been called.